src/storage/src/hummock/store/version.rs

```diff
@@ -61,6 +62,10 @@ use crate::store::{gen_min_epoch, ReadOptions, StateStoreIterExt, StreamTypeOfIt
 
 pub type CommittedVersion = PinnedVersion;
 
+/// The value of threshold should be related to the number of in-flight barriers
+/// and the frequency of checkpoint. We may tune this value later.
+pub const IMM_MERGE_THRESHOLD: usize = 8;
```

Comment:
Refactored to configurable in `StorageConfig`.
Yes. The current strategy to trigger imm merge is straightforward, and I have observed that data size (operator states) of each epoch can be different. We can optimize the strategy further. But pick up a reasonable size threshold is also not trivial.

Reply:
> In the current implementation, a merged IMM canont be merged again so we can still have performance issue if we have many small merged IMMs.

Hmm, I think a merged IMM could merge again. We do the merge for each table shard and obey the asc order of epochs, we can still append a merged_imm to the `imms_by_table_shard` container, then it can be merged with imms generated by newer epochs. Am I miss something?